RAG_supporters/clustering/centroid_comparison.py

```diff
@@ -0,0 +1,267 @@
+"""
+Centroid comparison utilities for cluster-based similarity search.
+
+This module provides tools for comparing embeddings against cluster centroids:
+- Distance computation (Euclidean and Cosine)
+- Nearest cluster finding
+- Keyword and text comparison against centroids
+
+Useful for assigning new data to existing clusters or performing cluster-based
+retrieval and steering operations.
+
+TODO: Future extensions could include:
+- Additional distance metrics (Manhattan, Mahalanobis)
+- Batch comparison utilities for efficiency
+- Integration with steering mechanisms for neural pipeline control
+"""
+
+import json
+import logging
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+
+LOGGER = logging.getLogger(__name__)
+
+
+class CentroidComparator:
```

Comment:
Consider if not should be merged with KeywordClusterer from keyword_clustering.py